fiftyone/utils/labelstudio.py

```diff
@@ -0,0 +1,835 @@
+"""
+Utilities for working with annotations in
+`Label Studio <https://labelstud.io>`_.
+
+Run Label Studio:
+    1. As a python package:
+        pip install label-studio
+        label-studio start
+    2. As a docker image:
+        docker run -d -p 8080:8080 \
+            --name labelstudio \
+            -v /tmp/labelstudio:/data \
+            labelstudio/labelstudio
+
+Set environment variables:
+    export FIFTYONE_LABELSTUDIO_URL=<url>
+    export FIFTYONE_LABELSTUDIO_TOKEN=<token>
+
+Annotate:
+    dataset = fo.load_dataset("quickstart")
+    dataset.annotate(<anno_key>, label_field=<field_name>,
+                    label_type=<label_type>,
+                    classes=[list, of, classes],
+                    backend="labelstudio")
+    # Note: a new project will be created with a labelling config
+    # that matches label type and classes.
+    # Note: it might take a while because files have to be
+    # uploaded to the Label Studio server.
+
+Load annotations:
+    dataset.load_annotations(<anno_key>)
+    # Note: this will only load the latest annotation per task
+    # if multiple annotations exist for a task.
+"""
+import itertools
+import json
+import logging
+import os
+import random
+import string
+import webbrowser
+from copy import deepcopy
+from datetime import datetime as dt
+from typing import List
+
+import numpy as np
+
+import fiftyone as fo
+import fiftyone.core.utils as fou
+import fiftyone.core.labels as fol
+import fiftyone.utils.annotations as foua
+
+ls = fou.lazy_import(
+    "label_studio_sdk", callback=lambda: fou.ensure_import("label_studio_sdk")
+)
+
+logger = logging.getLogger(__name__)
+
+
+class LabelStudioBackendConfig(foua.AnnotationBackendConfig):
+    """Base class for Label Studio backend configuration.
+
+    Args:
+        name: the name of the backend
+        label_schema: a dictionary containing the description of label fields,
+            classes and attribute to annotate
+        media_field ("filepath"): string field name containing the paths to
+            media files on disk to upload
+        url: the URL of the Label Studio server
+        api_key: the API key to use for authentication
+        project_name: the name of the project to use on the Label Studio server
+
+    """
+
+    def __init__(
+        self,
+        name: str,
+        label_schema: str,
+        media_field="filepath",
+        url="http://localhost:8080",
```

Comment:
The default `url` can be set to `None` since it will already have a default value provided through `config.py` and this default value will not be used.

Reply:
I left the home page for LS as the default address however it will not lead to anywhere. if running locally users have to set localhost explicitely